packages/thirdweb/src/chainsaw/types.ts

```diff
@@ -0,0 +1,141 @@
+import type { Transaction as ViemTransaction } from "viem";
+import type { Hex } from "../utils/encoding/hex.js";
+import type { NFT as ParsedNFT } from "../utils/nft/parseNft.js";
+import type { Prettify } from "../utils/type-utils.js";
+
+export type ChainsawResponse<T = unknown> = {
+  data?: T;
+  error?: string;
+  page?: number;
+};
+
+export type ChainsawPagingParams = {
+  /**
+   * Number of items per page
+   */
+  pageSize?: number;
+  /**
+   * Page number
+   */
+  page?: number;
+};
+
+export type Block = {
+  hash: Hex;
+  blockNumber: number;
+  time: Date;
+  parentHash: Hex;
+  miner: Hex;
+  nonce: Hex;
+  baseFeePerGas?: string;
+  difficulty: string;
+  gasLimit: string;
+  gasUsed: string;
+  chainId: number;
+  version: number;
+};
+
+export type ChainsawBlock = {
+  number: bigint | null;
+  hash: Hex | null;
+  nonce: Hex | null;
+  logsBloom: Hex | null;
+  baseFeePerGas: bigint | null;
+  blobGasUsed: bigint;
+  difficulty: bigint;
+  excessBlobGas: bigint;
+  extraData: Hex;
+  gasLimit: bigint;
+  gasUsed: bigint;
+  miner: string;
+  mixHash: Hex;
+  parentHash: Hex;
+  receiptsRoot: Hex;
+  sealFields: Hex[];
+  sha3Uncles: Hex;
+  size: bigint;
+  stateRoot: Hex;
+  timestamp: bigint;
+  totalDifficulty: bigint | null;
+  transactionsRoot: Hex;
+  uncles: Hex[];
+  withdrawalsRoot?: Hex;
+};
+
+export type Event = {
+  name: string;
+  count: string;
+  time?: string;
+  chainId?: number;
+  contractAddress?: Hex;
+  args?: string;
+};
+
+export type Events = Event[];
+
+export type ChainsawEvent = Prettify<
+  Omit<Event, "count"> & {
+    count: bigint;
+  }
+>;
+
+export type ChainsawEvents = ChainsawEvent[];
+
+export type DecodedTransaction = {
+  functionName: string;
+  args?: string;
+};
+
+export type Transaction = {
```

Comment:
What are these types based on? Happy to adapt our existing types to be compatible with chainsaw/viem if possible. Ping me if you wanna chat about how we can do it. We should avoid duplicating types though

Reply:
This is what comes from Chainsaw API. We can never have the same type without doing some formatting since you cannot send bigints over http e.g., so they need to be converted from string.

Also I don't want to confine Chainsaw to existing types in Viem in case we want to start enriching the data with more stuff. I would rather convert it here in the SDK and work with types as close as possible to what you already have.

Maybe naming it `Transaction` causes confusion and should be renamed to something less generic

Reply:
we should _do_ the formatting to take the chainsaw data -> existing types

Reply:
Done in [cf02c9d](https://github.com/thirdweb-dev/js/pull/4004/commits/cf02c9daf9593fa052e33da9971f74da3bb32fa0) for Block, NFTs and Transactions

Reply:
Also did logs as close as possible [788efee](https://github.com/thirdweb-dev/js/pull/4004/commits/788efeeb9e9dbb269ad420c2af61b97b6d2ded2f)